libnetdata/queue/queue.c

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#include "queue.h"
+
+/*
+ * @Input:
+ *      q / The queue instance that the item will be pushed into
+ * 		i / The item which will be pushed into the queue
+ * @Return
+ *      If the node is pushed the queue, it returns with 1, otherwise 0 
+ */
+int queue_push(queue_t q, void* i)	
+{		
+	node temp;
+	temp = (node) malloc(sizeof(struct node_s));
+	if (temp == NULL){
+		return 0;
+	}
+	pthread_mutex_lock(&q->lock);
+	while(q->count == q->max){
+		if(q->blocked == 0){
+			pthread_mutex_unlock(&q->lock);
+			return 0;
```

Comment:
`temp` was allocated and not released in this case

Reply:
It is fixed with this [commit](https://github.com/netdata/netdata/pull/12556/commits/2e0a26332968144ec858907f857f9d62917e8ddc).